### ISSUE TITLE
Supervised and untrained encoder

### ISSUE THREAD

**Issue by likith012** [ycq091044/ContraWR#2]
Can you please provide some information about how the supervised model and the untrained encoder are trained in the paper? My assumption is that the supervised model is trained on both `pretext group + train group` and the encoder model is trained on only the `train` group. When I did run the `supervised.py` script I was getting an accuracy of `78.28` and a kappa score of only `0.3871`, clearly, the model is performing badly due to fewer training subjects and I was using the same random seed as yours. I have been experimenting on sleepEDF dataset.

And in the paper, why there is a difference in metrics between the table values and graph values. Do they point to the same results?

![image](https://user-images.githubusercontent.com/14799306/141761950-10dcfe15-3077-44f7-a3c1-6563dc6fd957.png)
![image](https://user-images.githubusercontent.com/14799306/141762122-5c1080b9-5575-44ff-abac-f035cecaf01a.png)

**Comment by ycq091044** [comment]
Let me provide a quick clarification, will check the results carefully later.

First, the **Supervised model** is trained on training data and tested on test data (remember the pre-text data has no labels, we cannot use them). The **Untrained model** is just an initialized encoder (have not been trained yet, while MoCo, SimCLR, BYOL, SimSiam, ContraWR are trained on pre-text data), then we trained an LR model (using the training data) on top of the **Untrained encoder** to do downstream classification.

Second, the percentage of training data is 5% in the table, while in the figure, we vary the training percentage. You will have to match the percentage of training data and then compare. Also, we use different random seeds to split the data for these two experiments. So, there might be minor discrepancies though they are both under 5% of training data.

---------

Can you please also run other models to compare the results? 78.28% is quite low for the supervised model. If that is indeed the case, then I suspect the performance of all methods should drop while their relative differences will still remain.

**Comment by likith012** [comment]
Thanks for the response, sure will run more runs on self-supervised code and will let you know. I am a bit confused about the byol loss you are using, are you using `dot product` similarity instead of the original byol papers `l2 loss` and it's not a symmetric loss as used in the original paper [loss func](https://github.com/jramapuram/BYOL/blob/5ea487e7304df0e0983e8fc6c01bf74a4ce3ce36/objective.py#L6)? Also I observed that the loss is summed up instead of averaged(although learning rate handles this). Have you tried using the original loss?

**Comment by likith012** [comment]
can you comment on above?

**Comment by ycq091044** [comment]
First, the dot product loss is equivalent to l2 loss given that the rep. vector is unit-norm. Second, sum or mean operator does not matter since lr can handle it. Third, I just looked at the link to the BYOL repo, they are also using dot product loss and sum operator, right? It should not be a big deal. 

Let me know whether you can improve the accuracy by modifying my codes. Thanks.
